chancepy/Basics.py

```diff
@@ -39,9 +40,34 @@ def letter(cls, casing: str = "lower") -> str:
         return cls.character(pool=pools[casing])
 
     @classmethod
-    def string(cls, length: Optional[int] = None, pool: Optional[str] = None) -> str:
-        if not length:
-            length = cls.natural(5, 20)
+    def string(cls, length: Optional[int] = None, pool: Optional[str] = None,
+               min_length: Optional[int] = None, max_length: Optional[int] = None) -> str:
+
+        def sample_length(_length, _min_length, _max_length) -> int:
+            # exact length specified
+            if _length is not None:
+                return _length
+
+            # no desired length specified: return reasonable size string (5-20) chars
+            if [_length, _min_length, _max_length] == [None] * 3:
+                return cls.natural(5, 20)
+
+            # return length between lower and upper bound (or if only 1 bound exists
+            # default min will be 0 and default max 1M)
+            _min_length = _min_length or 0
+            _max_length = _max_length or 1_000_000
```

Comment:
isn't this pretty high as a default value?
also - I'm pretty sure this would bug out if I pass (1M + 1) as min value :P
I think it's best to go _max = _min + 20, or something.

Reply:
it wouldn't bug because I'm raising ValueError in this case

changing default max to min*2